tests/unit/sidecars/live_data_watcher_loader_tests.py

```diff
@@ -0,0 +1,125 @@
+import json
+import os
+
+from unittest import mock
+
+import boto3
+import botocore.session
+import pytest
+
+from botocore.stub import Stubber
+from moto import mock_s3
+
+from baseplate.sidecars.live_data_watcher import _load_from_s3
+from baseplate.sidecars.live_data_watcher import _parse_loader_type
+from baseplate.sidecars.live_data_watcher import LoaderException
+from baseplate.sidecars.live_data_watcher import LoaderType
+from baseplate.sidecars.live_data_watcher import NodeWatcher
+
+
+@pytest.fixture()
+def start_mock_s3():
+    with mock_s3():
+        yield
+
+
+@pytest.fixture()
+def s3_stub(start_mock_s3):
+    s3 = botocore.session.get_session().create_client("s3")
+    with Stubber(s3) as stubber:
+        yield stubber
+
+
+@pytest.mark.parametrize(
+    "data,return_value",
+    (
+        # Non-obviously-PASSTHROUGH configs
+        (json.dumps({"live_data_watcher_load_type": "S3"}).encode("utf-8"), LoaderType.S3),
+        (
+            json.dumps({"live_data_watcher_load_type": "NOTS3"}).encode("utf-8"),
+            LoaderType.PASSTHROUGH,
+        ),
+        # Everything else should be PASSTHROUGH
+        (b"", LoaderType.PASSTHROUGH),
+        (json.dumps({"key": "value"}).encode("utf-8"), LoaderType.PASSTHROUGH),
+        (json.dumps([]).encode("utf-8"), LoaderType.PASSTHROUGH),
+        (json.dumps(1234).encode("utf-8"), LoaderType.PASSTHROUGH),
+        (json.dumps(None).encode("utf-8"), LoaderType.PASSTHROUGH),
+        (b"hey{look}at--this//ugly**string@of(ugly^^chars", LoaderType.PASSTHROUGH),
+    ),
+)
+def test_parse_loader_type(data, return_value):
+    assert _parse_loader_type(data) == return_value
+
+
+@pytest.mark.parametrize(
+    "missing_config",
+    (
+        "region_name",
+        "bucket_name",
+        "file_key",
+        "sse_key",
+    ),
+)
+def test_load_from_s3_missing_config(start_mock_s3, missing_config):
+    data = {
+        "region_name": "some_region",
+        "bucket_name": "some_bucket",
+        "file_key": "some_file",
+        "sse_key": "some_key",
+    }
+    data.pop(missing_config)
+
+    with pytest.raises(LoaderException):
+        assert _load_from_s3(json.dumps(data).encode("utf-8"))
+
+
+def test_successful_load_from_s3(start_mock_s3):
```

Comment:
I suppose this could also be converted to Stubber from moto, @KTAtkinson do you have strong feelings?

Reply:
II would prefer using `Stubber` because it will catch misuses of the API. The way it works now, it assumes that any method or attribute you use on the client or it's return values is valid.

Reply:
Alright, stubber is in place but we still need some mocking.

Because we create a client based on the znode data, we have to create it at runtime. because we create it at runtime, we can't inject our stubbed session/client easily. So I mock out the client create call to always return our stubbed session/client and then we can use stubber everywhere